exercises/palindrome/src/example.c

```diff
@@ -0,0 +1,75 @@
+#include "palindrome.h"
+#include <stdio.h>
+#include <assert.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdbool.h>
+
+/* Return true if the product of a and b is a palindrome, false otherwise */
+bool isPalindromeProduct(int a, int b)
+{
+   char number[10];
+   int length;
+
+   sprintf(number, "%d", a * b);
+   length = strlen(number);
+
+   for (int i = 0; i < length / 2; i++) {
+      if (number[i] != number[length - i - 1]) {
+         return false;
+      }
+   }
+   return true;
+}
+
+/* return the factors of a palindrome */
+Pair *getPalindromeFactors(int palindrome)
+{
+   Pair *tableP;
+   int sizeTableP = 0, tablePcount = 0;
+
+   /* Compute the size of the array */
+   for (int i = 1; i <= palindrome; i++) {
+      for (int j = 1; j <= palindrome; j++) {
+         if (i * j == palindrome) {
+            sizeTableP++;
+         }
+      }
+   }
+   assert((tableP = malloc(sizeTableP * sizeof(Pair))) != NULL);
```

Comment:
This is problematic. If you compiled a release build of this file you'd never allocate anything.

When you turn off asserts for a release build, the expression contained in it never executes, so the malloc would never be run. It's best to avoid doing anything inside an assert that has side-effects.

It's probably best to allocate the memory and return null if it fails.